src/routes/auth.py

```diff
@@ -1,65 +1,86 @@
-router = APIRouter()
+router = APIRouter(prefix='/auth', tags=["auth"])
+security = HTTPBearer()
 
 oauth2_scheme = OAuth2PasswordBearer(tokenUrl="/auth/login")
 
 
 @router.post(
-    "/auth/signup",
+    "/signup",
     response_model=UserResponseModel,
     status_code=status.HTTP_201_CREATED,
 )
-async def signup(user: UserCreateModel):
-    # Simulate user registration logic
-    new_user = UserDbModel(
-        id=1,
-        username=user.username,
-        email=user.email,
-        created_at=datetime.now(),
-        role=RoleEnum.user,
-        is_active=True,
-    )
-    return UserResponseModel(
+async def signup(user: UserCreateModel, db: Session = Depends(get_db)):
+    exist_user = await repository_users.get_user_by_email(user.email, db)
+    if exist_user:
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail="Account already exists")
+
+    user_count = await repository_users.get_user_count(db)
+    if user_count == 0:
+        user.role = RoleEnum.admin
+    else:
+        user.role = RoleEnum.user
+    user.hashed_password = auth_service.get_password_hash(user.hashed_password)
+    new_user = await repository_users.create_user(user, db)
+
+    response_data = UserResponseModel(
         user=new_user,
         role=new_user.role,
-        detail=f"A new '{new_user.role.name}' has been successfully created",
+        detail="User successfully created"
     )
+    return response_data
 
 
-@router.post("/auth/login", response_model=TokenModel)
-async def login_user(user: UserLoginModel):
-    # Simulate login and JWT generation
-    if user.email == "admin@example.com" and user.password == "fakepass":
-        access_token = await auth_service.create_access_token(
-            data={"sub": user.email, "role": "admin"}
-        )
-        refresh_token = await auth_service.create_refresh_token(
-            data={"sub": user.email, "role": "admin"}
-        )
-        return {
-            "access_token": access_token,
-            "refresh_token": refresh_token,
-            "token_type": "bearer",
-        }
-    raise HTTPException(
-        status_code=status.HTTP_401_UNAUTHORIZED, detail="Invalid credentials"
-    )
+@router.post(
+    "/login",
+    response_model=TokenModel
+)
+async def login_user(user: OAuth2PasswordRequestForm = Depends(), db: Session = Depends(get_db)):
+    user_login = await repository_users.get_user_by_email(user.username, db)
+    if user_login is None:
+        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="Invalid email")
+    if not auth_service.verify_password(user.password, user_login.hashed_password):
+        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="Invalid password")
+
+    access_token = await auth_service.create_access_token(data={"sub": user_login.email})
+    refresh_token = await auth_service.create_refresh_token(data={"sub": user_login.email})
+    await repository_users.update_token(user_login, refresh_token, db)
```

Comment:
Before this I would verify if the token has expired already or not

Reply:
Thanks for the suggestion. Will add that later!